src/lib.rs

```diff
@@ -677,4 +679,138 @@ mod tests {
         let bytes = [0, 1, 0, 2, 0, 3, 0];
         let _: Vec<u16> = Decode::decode(&bytes[..]).unwrap();
     }
+
+    #[test]
+    fn test_encode_bool() {
+        let value: bool = true;
+        let bytes = value.encode().unwrap();
+        assert_eq!(bytes.as_slice(), &[1]);
+    }
+
+    #[test]
+    fn test_decode_bool_true() {
+        let bytes = vec![1];
+        let decoded_value: bool = Decode::decode(bytes.as_slice()).unwrap();
+        assert_eq!(decoded_value, true);
+    }
+
+    #[test]
+    fn test_decode_bool_false() {
+        let bytes = vec![0];
+        let decoded_value: bool = Decode::decode(bytes.as_slice()).unwrap();
+        assert_eq!(decoded_value, false);
+    }
+
+    #[test]
+    #[should_panic(expected = "Unexpected byte 42")]
+    fn test_decode_bool_bail() {
+        let bytes = vec![42];
+        let _: bool = Decode::decode(bytes.as_slice()).unwrap();
+    }
+
+    #[test]
+    fn test_default_decode() {
+        struct Foo {
+            bar: u8,
+        }
+
+        impl Decode for Foo {
+            fn decode<R: Read>(input: R) -> Result<Self> {
+                Ok(Foo { bar: 42 })
+            }
+        }
+
+        let bytes = vec![42, 12, 68];
+        let mut foo: Foo = Foo { bar: 41 };
+        foo.decode_into(bytes.as_slice()).unwrap();
+        assert_eq!(foo.bar, 42);
+    }
+
+    #[test]
+    fn test_option_encode_into() {
+        let option = Some(0x12u8);
+        let mut vec: Vec<u8> = vec![];
+        option.encode_into(&mut vec).unwrap();
+        assert_eq!(vec, vec![1, 18]);
+    }
+
+    #[test]
+    fn test_option_none_encode_into() {
+        let option: Option<u8> = None;
+        let mut vec: Vec<u8> = vec![];
+        option.encode_into(&mut vec).unwrap();
+        assert_eq!(vec, vec![0]);
+    }
+
+    #[test]
+    #[should_panic(expected = "Unexpected byte 42")]
+    fn test_bail_option_decode_into() {
+        let mut option: Option<u8> = Some(42);
+        let bytes = vec![42];
+        option.decode_into(bytes.as_slice()).unwrap();
+    }
+
+    #[test]
+    fn test_some_option_decode_into() {
+        let mut option: Option<u8> = Some(42);
```

Comment:
Probably should change the initial value here to something other than 42 - otherwise if something was broken and `decode_into` was a no-op, the assertion would still pass.

Reply:
Good call. Updated this.